database/src/migrations/20240310163000_observation_sign.ts

```diff
@@ -0,0 +1,104 @@
+import { Knex } from 'knex';
+
+/**
+ * Create new tables with initial data:
+ * - sign
+ *
+ * Update existing tables:
+ * - Add 'sign' column to subcount table
+ *
+ * @export
+ * @param {Knex} knex
+ * @return {*}  {Promise<void>}
+ */
+export async function up(knex: Knex): Promise<void> {
+  await knex.raw(`--sql
+  ----------------------------------------------------------------------------------------
+    -- Create observation subcount sign lookup table
+    ----------------------------------------------------------------------------------------
+    SET search_path = biohub;
+
+    CREATE TABLE observation_subcount_sign (
+      observation_subcount_sign_id                  integer          GENERATED ALWAYS AS IDENTITY (START WITH 1 INCREMENT BY 1),
+      name                                          varchar(32)        NOT NULL,
+      description                                   varchar(256),
+      record_end_date                               timestamptz(6),
```

Comment:
We should add a record_end_date constraint:

`CREATE UNIQUE INDEX observation_subcount_sign_nuk1 ON observation_subcount_sign(name, (record_end_date is NULL)) where record_end_date is null;`

We've dubbed them `nuk` indexes (null unique key). 

It basically just says that the `name` must be unique IF the record_end date is null. This allows old end-dated records to have duplicate `name`s, but doesn't allow current records to have the same name.

Reply:
Fixed